### DISCUSSION TITLE
Is there a way to add an already prepared dataset for training?

### DISCUSSION THREAD

**Discussion by Dioskurides** [JarodMica/ai-voice-cloning#50] [General]
Hi, just as the title says, i already got my dataset formatted, i would simply like to train it, is that possible?

**Comment by WillIsback** [comment]
Hi, I was in the same case as you are. It is but it need at least to be structured like this:
`YourFormattedDataSet/
 - train.txt ----------- FILE **file where 'audio_file_name | transcription' are written. 1 entry = 1 audio segment
 - SpeakerFolder ----DIR`
    - Audiofolder -----DIR ***It can be wav or whaterver but it is the repertory where all your segmented audio are.*** 
       - audio_file_name.wav ---- FILE
`

to be sure i did everything as tortoise training script expect I formated a little audio sample (3mn) with the embed tool. Then I look how train.txt was written. Made a script to write and fill my own train.txt for my dataset exactly the same way the embed tool did.

When everything is done, go on the web-ui training, you skip tab 1 and go directly to tab 2. Select your dataset_formatted repertory as your dataset and apply your training settings.

Finally you import your training settings and run the training.

**Reply by Dioskurides** [reply]
Thank you ill try that.

**Reply by JarodMica** [reply]
Pretty much exactly what Willlsback said, it allows you to skip the prepare dataset part and jump straight to generate configuration.

One thing I might add on is that you should down sample to 22050hz for your audio samples so that it's in alignment with what tortoise is expecting.  

Else, the sequence is going to be too long and you may not get a good result.  I haven't tried training with another SR so this is speculation based on the architecture so I might try just to confirm sometime later.

**Reply by Dioskurides** [reply]
Im currently having another problem which keeps me from even trying i prepared a small test dataset via ui just to confirm if the problem exists out of the box and it does. I opened an issue
